test/parallel/test-url-parse-invalid-input.js

```diff
@@ -16,3 +16,5 @@ const url = require('url');
 ].forEach(function(val) {
   assert.throws(function() { url.parse(val); }, TypeError);
 });
+
+assert.throws(function() { url.parse('http://%E0%A4%A@fail'); }, URIError);
```

Comment:
Can you change the last parameter to `assert.throws()` to `/^URIError: URI malformed$/`.